src/CppUTestExt/MockSupport_c.cpp

```diff
@@ -382,7 +384,7 @@ MockExpectedCall_c* withConstPointerParameters_c(const char* name, const void* v
 
 MockExpectedCall_c* withFunctionPointerParameters_c(const char* name, void (*value)())
 {
-    expectedCall = &expectedCall->withParameter(name, value);
+    expectedCall = &expectedCall->withParameter(name, (Cpp_Func*)value);
```

Comment:
Apologies. Spoke too soon. Had to edit above comment.

A (void(*)()) cast in C++ context, unfortunately, does not cast away the C-ness. Only a typedef does that. I knew I had tried this one before the typedef -- what don't I do to please BV ;-)